libfreerdp/gdi/video.c

```diff
@@ -140,20 +140,20 @@ void gdi_video_control_uninit(rdpGdi* gdi, VideoClientContext* video)
 	gdi->video = NULL;
 }
 
-static void gdi_video_timer(void* ctx, TimerEventArgs* timer)
+static void gdi_video_timer(rdpContext *ctx, TimerEventArgs* timer)
 {
-	rdpGdi* gdi = (rdpGdi*)ctx;
+	rdpGdi* gdi = ctx->gdi;
 
 	if (gdi && gdi->video)
 		gdi->video->timer(gdi->video, timer->now);
 }
 
 void gdi_video_data_init(rdpGdi* gdi, VideoClientContext* video)
 {
-	PubSub_SubscribeTimer(gdi->context->pubSub, gdi_video_timer);
+	PubSub_SubscribeTimer(gdi->context->pubSub, (pTimerEventHandler)gdi_video_timer);
```

Comment:
These casts are quite bad as the compiler will no longer warn if something in the `PubSub_SubscribeTimer` arguments changes

Reply:
Well I will change them, but nothing can change in the arguments as these functions are automatically generated by some macros.